packages/expo/src/winter/TextDecoder.ts

```diff
@@ -0,0 +1,422 @@
+    input.buffer instanceof ArrayBuffer
+  ) {
+    return new Uint8Array(input.buffer, input.byteOffset, input.byteLength);
+  }
+  return new Uint8Array(0);
+}
+
+/**
+ * End-of-stream is a special token that signifies no more tokens
+ * are in the stream.
+ */
+const END_OF_STREAM = -1;
+
+const FINISHED = -1;
+
+/**
+ * A stream represents an ordered sequence of tokens.
+ *
+ * @constructor
+ * @param {!(number[]|Uint8Array)} tokens Array of tokens that provide the stream.
+ */
+class Stream {
+  private tokens: number[];
+
+  constructor(tokens: number[] | Uint8Array) {
+    this.tokens = Array.prototype.slice.call(tokens);
+    // Reversed as push/pop is more efficient than shift/unshift.
+    this.tokens.reverse();
+  }
+
+  /**
+   * @return {boolean} True if end-of-stream has been hit.
+   */
+  endOfStream(): boolean {
+    return !this.tokens.length;
+  }
+
+  /**
+   * When a token is read from a stream, the first token in the
+   * stream must be returned and subsequently removed, and
+   * end-of-stream must be returned otherwise.
+   *
+   * @return {number} Get the next token from the stream, or
+   * end_of_stream.
+   */
+  read(): number {
+    if (!this.tokens.length) return END_OF_STREAM;
+    return this.tokens.pop()!;
+  }
+
+  /**
+   * When one or more tokens are prepended to a stream, those tokens
+   * must be inserted, in given order, before the first token in the
+   * stream.
+   *
+   * @param token The token(s) to prepend to the stream.
+   */
+  prepend(token: number | number[]): void {
+    if (Array.isArray(token)) {
+      while (token.length) this.tokens.push(token.pop()!);
+    } else {
+      this.tokens.push(token);
+    }
+  }
+
+  /**
+   * When one or more tokens are pushed to a stream, those tokens
+   * must be inserted, in given order, after the last token in the
+   * stream.
+   *
+   * @param token The tokens(s) to push to the stream.
+   */
+  push(token: number | number[]): void {
+    if (Array.isArray(token)) {
+      while (token.length) this.tokens.unshift(token.shift()!);
+    } else {
+      this.tokens.unshift(token);
+    }
+  }
+}
+
+function decoderError(fatal: boolean, opt_code_point?: number) {
+  if (fatal) throw TypeError('Decoder error');
+  return opt_code_point || 0xfffd;
+}
+
+interface Encoding {
+  name: string;
+  labels: string[];
+}
+
+const LABEL_ENCODING_MAP: { [key: string]: Encoding } = {};
+
+function getEncoding(label: string): Encoding | null {
+  label = label.trim().toLowerCase();
+  if (label in LABEL_ENCODING_MAP) {
+    return LABEL_ENCODING_MAP[label];
+  }
+  return null;
+}
+
+/** [Encodings table](https://encoding.spec.whatwg.org/encodings.json) (Incomplete as we only need TextDecoder utf8 in Expo RSC. A more complete implementation should be added to Hermes as native code.) */
+const ENCODING_MAP: { heading: string; encodings: Encoding[] }[] = [
+  {
+    encodings: [
+      {
+        labels: ['unicode-1-1-utf-8', 'utf-8', 'utf8'],
```

Comment:
These are all in the spec. Did you intentionally omit some of the values or can they be included? They all should map to the same UTF-8 encoding.
```
"unicode-1-1-utf-8"
"unicode11utf8"
"unicode20utf8"
"utf-8"
"utf8"
"x-unicode20utf8"
```

Reply:
Good call